Tests/UnitTests/Core/Utilities/FpeMonitorTests.cpp

```diff
@@ -14,42 +15,235 @@
+  BOOST_CHECK_EQUAL(mon.result().stackTraces().size(), 1);
+
+  invalid();
+  // We can't observe this again because it's masked!
+  BOOST_CHECK_EQUAL(mon.result().count(FpeType::FLTINV), 1);
+  BOOST_CHECK_EQUAL(mon.result().stackTraces().size(), 1);
+
+  mon.rearm();
+  invalid();
+  BOOST_CHECK_EQUAL(mon.result().count(FpeType::FLTINV), 2);
+  BOOST_CHECK_EQUAL(mon.result().stackTraces().size(), 2);
+  mon.result().deduplicate();
+  // now 1 because we deduplicated!
+  BOOST_CHECK_EQUAL(mon.result().stackTraces().size(), 1);
+
+  mon.rearm();
+  invalid2();
+  BOOST_CHECK_EQUAL(mon.result().count(FpeType::FLTINV), 3);
+  BOOST_CHECK_EQUAL(mon.result().stackTraces().size(), 2);
+  mon.result().deduplicate();  // doesn't do anything here actually
+  BOOST_CHECK_EQUAL(mon.result().stackTraces().size(), 2);
+}
+
+BOOST_AUTO_TEST_CASE(Scoping) {
+  {
+    FpeMonitor mon;
+    BOOST_CHECK(!mon.result().encountered(FpeType::FLTINV));
+    BOOST_CHECK(!mon.result().encountered(FpeType::FLTOVF));
+    BOOST_CHECK(!mon.result().encountered(FpeType::FLTDIV));
+
+    invalid();
+
+    {
+      FpeMonitor mon2;
+      BOOST_CHECK(!mon2.result().encountered(FpeType::FLTINV));
+      BOOST_CHECK(!mon2.result().encountered(FpeType::FLTOVF));
+      BOOST_CHECK(!mon2.result().encountered(FpeType::FLTDIV));
+
+      overflow();
+
+      {
+        FpeMonitor mon3;
+        BOOST_CHECK(!mon3.result().encountered(FpeType::FLTINV));
+        BOOST_CHECK(!mon3.result().encountered(FpeType::FLTOVF));
+        BOOST_CHECK(!mon3.result().encountered(FpeType::FLTDIV));
+
+        divbyzero();
+
+        BOOST_CHECK(!mon3.result().encountered(FpeType::FLTINV));
+        BOOST_CHECK(!mon3.result().encountered(FpeType::FLTOVF));
+        BOOST_CHECK(mon3.result().encountered(FpeType::FLTDIV));
+
+        // Test merging here
+        auto merged = mon.result().merge(mon2.result());
+        BOOST_CHECK_EQUAL(mon.result().count(FpeType::FLTINV), 1);
+        BOOST_CHECK_EQUAL(mon2.result().count(FpeType::FLTOVF), 1);
+        BOOST_CHECK_EQUAL(merged.count(FpeType::FLTINV), 1);
+        BOOST_CHECK_EQUAL(merged.count(FpeType::FLTOVF), 1);
+        BOOST_CHECK_EQUAL(merged.numStackTraces(), 2);
+        merged = merged.merge(mon3.result());
+        BOOST_CHECK_EQUAL(merged.count(FpeType::FLTINV), 1);
+        BOOST_CHECK_EQUAL(merged.count(FpeType::FLTOVF), 1);
+        BOOST_CHECK_EQUAL(merged.count(FpeType::FLTDIV), 1);
+        BOOST_CHECK_EQUAL(merged.numStackTraces(), 3);
+      }
+
+      BOOST_CHECK(!mon2.result().encountered(FpeType::FLTINV));
+      BOOST_CHECK(mon2.result().encountered(FpeType::FLTOVF));
+      BOOST_CHECK(!mon2.result().encountered(FpeType::FLTDIV));
+    }
+
+    BOOST_CHECK(mon.result().encountered(FpeType::FLTINV));
+    BOOST_CHECK(!mon.result().encountered(FpeType::FLTOVF));
+    BOOST_CHECK(!mon.result().encountered(FpeType::FLTDIV));
+  }
+}
+
+BOOST_AUTO_TEST_CASE(MergeDeduplication) {
+  {
```

Comment:
same as above - is this first block necessary?

Reply:
Removing